### DISCUSSION TITLE
How to convert direct rules into policy since they are not supported with nftables backend

### DISCUSSION THREAD

**Discussion by rgaduput** [firewalld/firewalld#1373] [Q&A]
Hi,
We have few direct rules which used to work fine with legacy iptables on CentOS7 but now since they are not supported in nftables (RHEL9 based OS), I am trying to convert those. tried several ways but no luck. 
Appreciate any suggestion on what am i doing wrong.

```
[root@master-2586-10497 ~]# firewall-cmd --version
1.3.4
[root@master-2586-10497 ~]#
[root@master-2586-10497 ~]# nft --version
nftables v1.0.9 (Old Doc Yak #3)
[root@master-2586-10497 ~]# 
```

Direct rules which worked on legacy
```
firewall-cmd --permanent --direct --add-rule ipv4 filter FORWARD 99 -o cali+ -j ACCEPT
firewall-cmd --permanent --direct --add-rule ipv4 filter FORWARD 99 -i cali+ -j ACCEPT
```
What I have tried

```
firewall-cmd --permanent --new-zone=kubeAccept
firewall-cmd --permanent --zone=kubeAccept --set-target=ACCEPT
firewall-cmd --permanent --zone=kubeAccept --add-interface="cali+"
firewall-cmd --reload
```

```
firewall-cmd --permanent --new-zone cali
firewall-cmd --permanent --zone cali --add-interface cali+
firewall-cmd --permanent --zone cali --set-target ACCEPT
firewall-cmd --permanent --new-policy allowAllFwd
firewall-cmd --permanent --policy allowAllFwd --add-ingress-zone cali
firewall-cmd --permanent --policy allowAllFwd --add-egress-zone ANY
firewall-cmd --permanent --policy allowAllFwd --set-target ACCEPT
```

**Comment by erig0** [comment]
I see you discussion in #527 as well. In #527, then `nft monitor` trace shows a packet ingressing `eth0`. The policy you have above is allowing `cali+` --> `eth0`, i.e. the opposite direction.

Are you trying to allow pod to world trafcic? pod to pod on a different host? pod to pod locally?

**Reply by rgaduput** [reply]
I was trying to setup pod to pod communication with in the same node and across worker/master nodes in the cluster.

**Reply by erig0** [reply]
You're going to need a couple policies. One for each direction.

One for each of:

1. pod to other (pod, world, other node)
  - your existing policy, `allowAllFwd` should cover this
 
2. other node (or world) to pod
  - similar to your existing `allowAllFwd`, but with `egress-zone=cali+` and `ingress-zone=ANY`
  - this missing policy seems to be what you're seeing in #527
